torch/csrc/jit/runtime/static/passes.cpp

```diff
@@ -950,6 +931,149 @@ void FuseListUnpack(std::shared_ptr<torch::jit::Graph>& graph) {
+  const auto equally_split = fromQualString("fb::equally_split");
+  const auto get_item = fromQualString("aten::__getitem__");
+  for (auto node = graph_it.next(); node != nullptr; node = graph_it.next()) {
+    if (node->kind() != equally_split) {
+      continue;
+    }
+
+    const Value* value_out = node->outputs()[0];
+
+    const auto new_sym = fromQualString("static_runtime::fused_equally_split");
+
+    const Value* value_num_splits = node->inputs()[1];
+
+    // check if equally_split num of splits is equal to the number of output uses
+    if(!toIValue(value_num_splits).has_value() || !toIValue(value_num_splits).value().isInt() || toIValue(value_num_splits).value().toInt() != value_out->uses().size()) {
+      continue;
+    }
+
+    bool should_fuse = true;
+
+    std::vector<UserToRemove> user_to_remove_list;
+
+    std::unordered_set<std::int64_t> getItem_indices;
+
+    for(auto index = 0; index < value_out->uses().size(); index++) {
+      Node* user = value_out->uses()[index].user;
+
+      // check if all users are get_item. Even if one is not we cannot fuse
+      if (user->kind() != get_item)
+      {
+        should_fuse = false;
+        break;
+      }
+
+      auto getItem_indice_out = toIValue(user->inputs()[1]);
+
+      // check if there is duplicate get_item use which is same output node index accessed by different get item.
+      // If there is duplicate we cannot fuse
+      if(!getItem_indice_out.has_value() || getItem_indices.contains(getItem_indice_out.value().toInt())) {
+        should_fuse = false;
+        continue;
+      }
+      auto user_outputs = user->outputs();
+
+      if (user_outputs.empty()) {
+        should_fuse = false;
+        break;
+      }
+
+      LOG(INFO) << "Found fb::equally_split node2: " << user->kind().toQualString();
+
+      getItem_indice_out.emplace(getItem_indice_out);
+      UserToRemove user_to_remove;
+      user_to_remove.user = user;
+      user_to_remove.user_outputs = user->outputs();
+      user_to_remove_list.emplace_back(user_to_remove);
+    }
+
+
+    if (!should_fuse) {
+      continue;
+    }
+
+    Replacement replacement;
+    replacement.node_to_be_fused = node;
+    replacement.new_sym = new_sym;
+    replacement.user_to_remove = std::move(user_to_remove_list);
+    replacement_list.emplace_back(replacement);
+  }
+
+  for(const auto& replacement : replacement_list) {
+    auto* new_node = graph->create(replacement.new_sym, 0);
+    for (Value* in : replacement.node_to_be_fused->inputs()) {
+        new_node->addInput(in);
+      }
+    for(const auto& user_to_remove_node : replacement.user_to_remove) {
+      for (Value* out : user_to_remove_node.user_outputs) {
+        Value* new_out = new_node->addOutput();
+        new_out->copyMetadata(out);
+        out->replaceAllUsesWith(new_out);
+      }
+      user_to_remove_node.user->destroy();
+    }
+    new_node->insertAfter(replacement.node_to_be_fused);
+    replacement.node_to_be_fused->destroy();
+  }
```

Comment:
it would be easier for me to review this for correctness if it ended up looking the same as the FuseListUnpack path. If you need to send a separate stacked PR to get rid of the std::tuple in FuseListUnpack, go for it